### DISCUSSION TITLE
Prompt-based method that could be re-used in multiple Skills

### DISCUSSION THREAD

**Discussion by Rhastamagic** [microsoft/semantic-kernel#1478] [Q&A]
As far as I understand, we have two kinds of Skills in SK: code-based and prompt-based.
I am wondering whether SK has prompt-based methods that could be reused in multiple code-based Skills.

Provide a case to help me introduce what I want:
SK agent could generate the plan in terms of the goal I described.
When it executes the plan step by step, it may find some core information is missing in the goal and then needs to interact with the user to get this additional info.
I am seeking a way to implement this kind of 'interaction' that can be defined in prompt style and can be reused in multiple Skills.

**Comment by evchaki** [comment]
@Rhastamagic  check out the updated planner - this should help - https://devblogs.microsoft.com/semantic-kernel/migrating-from-the-sequential-and-stepwise-planners-to-the-new-handlebars-and-stepwise-planner/
